### PR TITLE
Shayan/feq 2183/implement account switcher

### PR DESCRIPTION
**Pull request by shayan-deriv**
Usage Example:
```
                        <AccountSwitcher
                            activeAccount={activeAccount}
                        >
                            <AccountSwitcher.Tab title="Real">

                                <AccountSwitcher.AccountsPanel
                                    isOpen
                                    title="EU Accounts"
                                >
                                    {accountsList
                                        .map((account) => (
                                            <AccountSwitcher.AccountsItem
                                                key={account.loginid}
                                                account={account}
                                                onSelectAccount={() => {}}
                                            />
                                        ))}
                                </AccountSwitcher.AccountsPanel>

                                <Divider color="#f2f3f4" height="4px" />

                                <AccountSwitcher.AccountsPanel title="Non-EU">
                                    {accountsList
                                        .map((account) => (
                                            <AccountSwitcher.AccountsItem
                                                key={account.loginid}
                                                account={account}
                                                onSelectAccount={() => {}}
                                            />
                                        ))}
                                </AccountSwitcher.AccountsPanel>

                                <Divider color="#f2f3f4" height="4px" />

                                <AccountSwitcher.TotalAsset
                                    title="Total assets"
                                    description="test description text comes here"
                                    value="10,021 USD"
                                />

                                <Divider color="#f2f3f4" height="4px" />

                                <AccountSwitcher.TradersHubLink>
                                    Looking for CFD?{" "}
                                </AccountSwitcher.TradersHubLink>

                                <Divider color="#f2f3f4" height="4px" />

                                <AccountSwitcher.Footer>
                                    this is a footer
                                </AccountSwitcher.Footer>

                            </AccountSwitcher.Tab>

                            <AccountSwitcher.Tab title="Demo">
                                test 2
                            </AccountSwitcher.Tab>

                        </AccountSwitcher>
```

Screen Record:
https://github.com/deriv-com/ui/assets/100833613/e86b5945-503a-4e80-9022-32578c640c90

### CODE REVIEW

File: src/components/AppLayout/AccountSwitcher/AccountSwitcher.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+.ReactModal__Overlay--after-open{
+    opacity: 1;
+    transition: opacity 200ms ease-in-out;
+}
+
+.ReactModal__Overlay--before-close{
+    opacity: 0;
+    transition: opacity 400ms ease-in-out;
+}
```
Comment (shayan-deriv) [inline src/components/AppLayout/AccountSwitcher/AccountSwitcher.scss:55]: these are for modal animation in mobile

File: src/components/AppLayout/AccountSwitcher/TotalAsset/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    description,
+    value,
+    className,
+    ...rest
+}: TotalAssetProps) => {
+    return (
+        <div className={clsx("deriv-total-assets", className)} {...rest}>
+            <div className="deriv-total-assets__details">
+                <span className="deriv-total-assets__title">{title}</span>
+                <span className="deriv-total-assets__description">
```
Comment (nada-deriv) [inline src/components/AppLayout/AccountSwitcher/TotalAsset/index.tsx:21]: i just had a doubt.. why are we not using the text component here?

Comment (shayan-deriv) [reply src/components/AppLayout/AccountSwitcher/TotalAsset/index.tsx:21]: I wasn't thinking about it. let me update the code 🙏

**Review by matin-deriv — APPROVED** [review]

File: src/components/AppLayout/AccountSwitcher/TotalAsset/TotalAsset.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    &__value{
+        text-align: right;
+        font-size: 14px;
+        color: var(--du-text-general, #333333);
+        font-weight: 400;
+    }
+
+
+}
```
Comment (nada-deriv) [inline src/components/AppLayout/AccountSwitcher/TotalAsset/TotalAsset.scss:34]: may be the prettier didn't run ..

**Review by nada-deriv — APPROVED** [review]

**Pull request merged**